x-pack/plugins/apm/common/agent_configuration/setting_definitions/java_settings.ts

```diff
@@ -47,6 +69,125 @@ export const javaSettings: RawSettingDefinition[] = [
+          'Setting to true will enable instrumentations in the experimental group.',
+      }
+    ),
+    includeAgents: ['java'],
+  },
+
+  {
+    key: 'enable_instrumentations',
+    type: 'text',
+    defaultValue: '',
+    label: i18n.translate(
+      'xpack.apm.agentConfig.enableInstrumentations.label',
+      {
+        defaultMessage: 'Disable instrumentations',
+      }
+    ),
+    description: i18n.translate(
+      'xpack.apm.agentConfig.enableInstrumentations.description',
+      {
+        defaultMessage:
+          'A list of instrumentations which should be selectively enabled. ' +
+          'Valid options are listed in the ' +
+          '[Java APM Agent documentation](https://www.elastic.co/guide/en/apm/agent/java/current/config-core.html#config-disable-instrumentations).\n' +
+          '\n' +
+          'When set to non-empty value, only listed instrumentations will be enabled ' +
+          'if they are not disabled through `disable_instrumentations`or `enable_experimental_instrumentations`.\n' +
+          'When not set or empty (default), all instrumentations enabled by default will be enabled ' +
+          'unless they are disabled through `disable_instrumentations` or `enable_experimental_instrumentations`.',
+      }
+    ),
+    includeAgents: ['java'],
+  },
+
+  {
+    key: 'log_sending',
+    type: 'boolean',
+    defaultValue: 'false',
+    label: i18n.translate('xpack.apm.agentConfig.logSending.label', {
+      defaultMessage: 'Log sending (experimental)',
+    }),
+    description: i18n.translate(
+      'xpack.apm.agentConfig.logSending.description',
+      {
+        defaultMessage:
+          'Experimental, requires latest version of the Java agent.\n' +
+          '\n' +
+          'If set to `true`,\n' +
+          'agent will send logs directly to APM server.',
+      }
+    ),
+    includeAgents: ['java'],
+  },
+
+  {
+    key: 'mongodb_capture_statement_commands',
+    type: 'boolean',
+    defaultValue: 'false',
+    label: i18n.translate(
+      'xpack.apm.agentConfig.mongodbCaptureStatementCommands.label',
+      {
+        defaultMessage: 'MongoDB capture statement commands',
+      }
+    ),
+    description: i18n.translate(
+      'xpack.apm.agentConfig.mongodbCaptureStatementCommands.description',
+      {
+        defaultMessage:
+          'MongoDB command names for which the command document will be captured, limited to common read-only operations by default. ' +
+          'Set to `""` (empty) to disable capture, and `*` to capture all (which is discouraged as it may lead to sensitive information capture).\n' +
+          '\n' +
+          'This option supports the wildcard `*`, which matches zero or more characters. Examples: `/foo/*/bar/*/baz*`, `*foo*`. ' +
+          'Matching is case insensitive by default. Prepending an element with `(?-i)` makes the matching case sensitive.',
+      }
+    ),
+    includeAgents: ['java'],
+  },
```

Comment:
Q: What decides if this entry should be in `java_settings.ts` vs. in `general_settings.ts`? The Node.js agent intends to implement this config var *eventually*.

Reply:
> What decides if this entry should be in java_settings.ts vs. in general_settings.ts? 

Actually, things like this:

> The Node.js agent intends to implement this config var eventually.

Will move it to general settings then.

Reply:
@AlexanderWert btw. feel free to refactor that separation (java vs general) if you want. Not sure if it still makes sense